### PR TITLE
Add additional helm documentation for Fleet Server, and Agent

### PR DESCRIPTION
**Pull request by naemono**
This adds additional documentation for installing/managing Fleet Server, and managed Elastic Agents using the Helm charts. 

While writing this, I noticed that the referenced examples in the documentation for Elasticsearch/Kibana were missing, and this also adds them where they need to be.

### CODE REVIEW

**Review by pebrc — COMMENTED** [review]
LGTM but I would make the paths branch version specific. While testing this I realised that our Helm Chart release did not work properly. The Elastic Stack chart I get is still the one prior to the 2.5 release. We did not bump the Chart version when we added new dependencies which is a mistake, which I think we can only fix through a new release.

File: docs/orchestrating-elastic-stack-applications/stack-helm-chart.asciidoc [hunk 1]
```diff
@@ -45,6 +45,18 @@ To use one or more of these example configurations, use the `--values` Helm opti
...
+[float]
+[id="{p}-install-fleet-agent-elasticsearch-kibana-helm"]
+== Installing Fleet Server with Elastic Agents along with Elasticsearch and Kibana using the eck-stack Helm Chart
+
+The following section builds upon the previous section, and allows installing Fleet Server, and Fleet-managed Elastic Agents along with Elasticsearch and Kibana.
+
+[source,sh]
+----
+# Install an eck-managed Elasticsearch, Kibana, Fleet Server, and managed Elastic Agents using custom values.
+helm install eck-stack-with-fleet elastic/eck-stack --values https://github.com/elastic/cloud-on-k8s/tree/main/deploy/eck-stack/examples/agent/fleet-agents.yaml -n elastic-stack
```
[the remark this answers is not in the document]
Comment (naemono) [reply docs/orchestrating-elastic-stack-applications/stack-helm-chart.asciidoc]: @pebrc The values links have all been updated.  The Helm chart release process issue is being handled separately.

File: deploy/eck-stack/examples/elasticsearch/hot-warm-cold.yaml [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+eck-elasticsearch:
+  nodeSets:
+  - name: masters
+    count: 1
+    config:
+      node.roles: ["master"]
+      # Comment out when setting the vm.max_map_count via initContainer, as these are mutually exclusive.
+      # For production workloads, it is strongly recommended to increase the kernel setting vm.max_map_count to 262144
+      # and leave node.store.allow_mmap unset.
+      # ref: https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-virtual-memory.html
```
Comment (thbkrkr) [inline deploy/eck-stack/examples/elasticsearch/hot-warm-cold.yaml:11]:
```suggestion
      # ref: https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-virtual-memory.html
```

File: deploy/eck-stack/examples/elasticsearch/hot-warm-cold.yaml [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+        #       nodeSelectorTerms:
+        #       - matchExpressions:
+        #         - key: beta.kubernetes.io/instance-type
+        #           operator: In
+        #           # This should be adjusted to the instance type according to your setup
+        #           #
+        #           values:
+        #           - highio
+    # Volume Claim settings.
+    # ref: https://www.elastic.co/guide/en/cloud-on-k8s/2.2/k8s-volume-claim-templates.html
```
Comment (thbkrkr) [inline deploy/eck-stack/examples/elasticsearch/hot-warm-cold.yaml:39]:
```suggestion
    # ref: https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-volume-claim-templates.html
```

**Review by thbkrkr — COMMENTED** [review]
Looks good, one last little thing: use `current` instead of `master` or `2.2` in the ECK doc URLs referenced in `deploy/eck-stack/examples/elasticsearch/hot-warm-cold.yaml` for consistency.

File: deploy/eck-stack/examples/elasticsearch/hot-warm-cold.yaml (hunk 2 above)
Comment (thbkrkr) [reply deploy/eck-stack/examples/elasticsearch/hot-warm-cold.yaml]: I haven't created suggestions for all occurrences but there are others, same for `k8s-volume-claim-templates.html`.

**Review by pebrc — APPROVED** [review]

**Comment by thbkrkr** [comment]
I resolved the conflict with `main`. Sorry for the late comments but this makes me ask new questions.

File: deploy/eck-stack/examples/agent/fleet-agents.yaml [hunk 4]
```diff
@@ -28,26 +24,18 @@ eck-kibana:
   #
   fullnameOverride: kibana
   
-  # Version of Kibana.
-  #
-  version: 8.5.0
-
```
Comment (thbkrkr) [inline deploy/eck-stack/examples/agent/fleet-agents.yaml:34]: Should we also remove the `version` defined for the Agent?

File: deploy/eck-stack/examples/agent/fleet-agents.yaml [hunk 5]
```diff
@@ -28,26 +24,18 @@ eck-kibana:
...
-  
     # Reference to ECK-managed Elasticsearch instance, ideally from {{ "elasticsearch.fullname" }}
     #
     elasticsearchRef:
       name: elasticsearch
 
     config:
       # Note that these are specific to the namespace into which this example is installed, and are
-      # using `default` as configured here.  If installed outside of the `default` namespace,
+      # using `elastic-stack` as configured here.  If installed outside of the `elastic-stack` namespace,
```
Comment (thbkrkr) [inline deploy/eck-stack/examples/agent/fleet-agents.yaml:35]: I was looking where is configured the `elastic-stack` namespace and IIUC it's from README.md:L40 when we explain how to use the chart `helm install es-kb-quickstart elastic/eck-stack -n elastic-stack --create-namespace`. Ok, but it's a bit tricky if you look only at the yaml file.

**Review by thbkrkr — APPROVED** [review]

**Comment by naemono** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|2.6|[](https://github.com/elastic/cloud-on-k8s/pull/6263)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**